base/abstractarray.jl

```diff
@@ -38,18 +38,12 @@ function first(itr)
 end
 last(a) = a[end]
 
-function stride(a::AbstractArray, i::Integer)
-    if i > ndims(a)
-        return length(a)
-    end
-    s = 1
-    for n=1:(i-1)
-        s *= size(a, n)
-    end
-    return s
-end
+strides{T}(a::AbstractArray{T,0}) = ()
+strides(a::AbstractArray) = _strides(a,(1,))
+_strides{_,N}(a::AbstractArray{_,N}, s::NTuple{N,Int}) = s
+_strides{_,N,M}(a::AbstractArray{_,N}, s::NTuple{M,Int}) = _strides(a, (s..., s[M]*size(a, M)))
 
-strides(a::AbstractArray) = ntuple(i->stride(a,i), ndims(a))::Dims
+stride(a::AbstractArray, d) = d <= 1 ? 1 : size(a, d-1)*stride(a, d-1)
```

Comment:
Indeed, it is still slower, though I hadn't thought it was a factor of two. I'll change it back to the old more lengthy definition. I guess it was to be expected that loops win over recursion. I just wanted to mimick the `strides` definition where recursive is the way to go to construct tuples in a type stable manner.